contracts/treasury/priceOracle/KyberDmmChainLinkPriceOracle.sol

```diff
@@ -0,0 +1,477 @@
+   *      liquidate them to a single whitelisted token
+   *  - LIQUIDATE_LP: Take list of LP tokens, then liquidate them to a single whitelisted token
+   *  Apply premium discount, can be a different value for each liquidator.
+   * @param liquidator address of the liquidator
+   * @param tokenIns list of src tokens
+   * @param amountIns list of src amounts
+   * @param tokenOuts list of return tokens
+   * @param hint hint for getting conversion rates
+   * @return minAmountOuts min expected amount for each token out
+   */
+  function getExpectedReturns(
+    address liquidator,
+    IERC20Ext[] calldata tokenIns,
+    uint256[] calldata amountIns,
+    IERC20Ext[] calldata tokenOuts,
+    bytes calldata hint
+  )
+    external override view
+    returns (uint256[] memory minAmountOuts)
+  {
+    require(tokenIns.length == amountIns.length, 'invalid length');
+    minAmountOuts = new uint256[](tokenOuts.length);
+
+    (OracleHintType hintType) = abi.decode(hint, (OracleHintType));
+    uint64 premiumBps;
+
+    // Remove Liquidity given a LP token
+    if (hintType == OracleHintType.REMOVE_LIQUIDITY) {
+      require(tokenIns.length == 1, 'invalid number token in');
+      require(tokenOuts.length == 2, 'invalid number token out');
+      (IERC20Ext[2] memory tokens, uint256[2] memory amounts) = getExpectedTokensFromLp(
+        address(tokenIns[0]), amountIns[0]
+      );
+      if (tokens[0] == tokenOuts[0]) {
+        require(tokens[1] == tokenOuts[1], 'invalid token out 1');
+        (minAmountOuts[0], minAmountOuts[1]) = (amounts[0], amounts[1]);
+      } else {
+        require(tokens[0] == tokenOuts[1], 'invalid token out 1');
+        require(tokens[1] == tokenOuts[0], 'invalid token out 0');
+        (minAmountOuts[0], minAmountOuts[1]) = (amounts[1], amounts[0]);
+      }
+      (premiumBps, ,) = getPremiumData(liquidator);
+      return _applyPremiumFor(minAmountOuts, premiumBps);
+    }
+
+    require(tokenOuts.length == 1, 'invalid number token out');
+    require(isWhitelistedToken(address(tokenOuts[0])), 'token out must be whitelisted');
+
+    // Liquidate list of LP tokens to a single dest token
+    if (hintType == OracleHintType.LIQUIDATE_LP) {
+      for(uint256 i = 0; i < tokenIns.length; i++) {
+        (IERC20Ext[2] memory tokens, uint256[2] memory amounts) = getExpectedTokensFromLp(
+          address(tokenIns[i]), amountIns[i]
+        );
+        // calc equivalent (tokens[0], amounts[2]) -> tokenOuts[0]
+        uint256 rate = conversionRate(address(tokens[0]), address(tokenOuts[0]), amounts[0]);
+        require(rate > 0, 'invalid conversion rate 0');
+        minAmountOuts[0] = minAmountOuts[0].add(
+          calcDestAmount(tokens[0], tokenOuts[0], amounts[0], rate)
+        );
+        // calc equivalent (tokens[1], amounts[3]) -> tokenOuts[0]
+        rate = conversionRate(address(tokens[1]), address(tokenOuts[0]), amounts[1]);
+        require(rate > 0, 'invalid conversion rate 1');
```

Comment:
I prefer `0 conversion rate`

Reply:
Furthermore, this is blocking us from liquidating token without chain-link oracle
E.g: Sushi-eth or rock-eth.

Reply:
for these pairs, they should call remove liquidity first, then liquidate with list of supported tokens